widget/embedded/src/pages/SwapDetails.tsx

```diff
@@ -0,0 +1,170 @@
+import { useUiStore } from '../store/ui';
+import { getPendingSwaps } from '../utils/queue';
+import { getFormatedPendingSwap } from '../utils/routing';
+import {
+  getLastConvertedTokenInFailedSwap,
+  getSwapMessages,
+  isNetworkStatusInWarningState,
+  shouldRetrySwap,
+} from '../utils/swap';
+import { getSwapDate } from '../utils/time';
+
+const RESET_INTERVAL = 2_000;
+
+export function SwapDetailsPage() {
+  const selectedSwapRequestId = useUiStore.use.selectedSwapRequestId();
+  const { canSwitchNetworkTo, connect, getWalletInfo } = useWallets();
+  const retry = useBestRouteStore.use.retry();
+  const navigate = useNavigate();
+  const { navigateBackFrom } = useNavigateBack();
+  const [isCopied, handleCopy] = useCopyToClipboard(RESET_INTERVAL);
+  const { manager, state } = useManager();
+
+  const pendingSwaps = getPendingSwaps(manager);
+  const selectedSwap = pendingSwaps.find(
+    ({ swap }) => swap.requestId === selectedSwapRequestId
+  );
+
+  const onCancel = () => {
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion, @typescript-eslint/no-non-null-asserted-optional-chain
+    const swap = manager?.get(selectedSwap?.id!);
+    if (swap) {
+      cancelSwap(swap);
+    }
+  };
+  const swap = selectedSwap?.swap;
+  const loading = !state.loadedFromPersistor;
+
+  if (!swap) {
+    return (
+      <SwapDetailsPlaceholder
+        requestId={selectedSwapRequestId || ''}
+        loading={loading}
+        onBack={navigateBackFrom.bind(null, navigationRoutes.swapDetails)}
+      />
+    );
+  }
+
+  const currentStep = getCurrentStep(swap);
+
+  const currentStepBlockchain = currentStep
+    ? getCurrentBlockchainOfOrNull(swap, currentStep)
+    : null;
+  const currentStepWallet = currentStep
+    ? getRelatedWalletOrNull(swap, currentStep)
+    : null;
+  const currentStepNetworkStatus = currentStep?.networkStatus;
+
+  const swapMessages = getSwapMessages(swap, currentStep);
+
+  const networkWarningState = isNetworkStatusInWarningState(currentStep);
+
+  const swapDate = getSwapDate(swap);
+
+  const shouldRetry = shouldRetrySwap(swap);
+
+  const isMobileWallet = (walletType: string): boolean =>
+    !!getWalletInfo(walletType)?.mobileWallet;
+
+  const showSwitchNetwork =
+    currentStepNetworkStatus ===
+      PendingSwapNetworkStatus.WaitingForNetworkChange &&
+    !!currentStepBlockchain &&
+    !!currentStepWallet?.walletType &&
+    (isMobileWallet(currentStepWallet.walletType) ||
+      canSwitchNetworkTo(currentStepWallet.walletType, currentStepBlockchain));
+
+  const switchNetwork = showSwitchNetwork
+    ? connect.bind(null, currentStepWallet.walletType, currentStepBlockchain)
+    : undefined;
+
+  const lastConvertedTokenInFailedSwap =
+    getLastConvertedTokenInFailedSwap(swap);
+
+  return (
+    <SwapHistory
+      onBack={navigateBackFrom.bind(null, navigationRoutes.swapDetails)}
+      /* TODO: It was temporarily removed to find a better solution*/
```

Comment:
I'm not in context of this TODO and commented coded. Please remove them if not needed anymore.

Reply:
the file is deleted.

Reply:
the file is deleted.